### PR TITLE
Password update endpoint

### PR DESCRIPTION
**Pull request by jmnda-dev**
- [x] Add serializer for changing user password.
- [x] Add endpoint for changing password
- [x] Register endpoint in API router
Screenshot from 2021-11-09 10-23-16

### CODE REVIEW

**Comment by sonlinux** [comment]
```
def profile_thumbnail(self.profile_photo):
    # process the image  here
    thumbnail =  f"{jachin_uuid_agent_code()[1].png} "  # jachin_34OUIERK
    return thumbnail
def  save(self):

    self.profile_photo = profile_thumbnail(self.profile_photo)
```

File: itez/users/api/serializers.py [hunk 1]
```diff
@@ -12,3 +13,59 @@ class Meta:
...
+
+        Raises:
+            serializers.ValidationError: This exception is raised if the old_password is not correct.
+
+        Returns:
+            [str]: Returns the value if the old_password is correct.
+        """
+        user = self.context['request'].user
+        if not user.check_password(value):
+            raise serializers.ValidationError({"old_password": "Old password is not correct"})
```
Comment (sonlinux) [inline itez/users/api/serializers.py:63]: just wondering here,are we showing the `old_passowrd` to the error message too?

File: itez/users/api/serializers.py [hunk 2]
```diff
@@ -12,3 +13,59 @@ class Meta:
...
+            attrs [collections.OrderedDict]: Contains data posted in the request body.
+
+        Raises:
+            [serializers.ValidationError]: An exception raise when password and password2 don't match.
+
+        Returns:
+            attrs [collections.OrderedDict]: Contains data posted in the request body.
+        """
+        if attrs['password'] != attrs['password2']:
+            raise serializers.ValidationError({"password": "Password fields didn't match."})
```
Comment (sonlinux) [inline itez/users/api/serializers.py:44]: just wondering here,are we showing the entered `password` in the error message too?

**Review by sonlinux — APPROVED** [review]
Beautiful work @jmnda-dev 

Just clarify for me the review thoughts and we can squash merge!

File: itez/users/api/serializers.py [hunk 3]
```diff
@@ -12,3 +13,59 @@ class Meta:
...
+
+        return attrs
+
+    def validate_old_password(self, value):
+        """
+        Validates the old_password provided is correct.
+
+        Args:
+            value ([str]): Data from the old_password field
```
Comment (sonlinux) [inline itez/users/api/serializers.py]: Lets have a full stop/dot at end of this docstring.

File: itez/users/api/serializers.py (hunk 1 above)
Comment (jmnda-dev) [reply itez/users/api/serializers.py:63]: Yes, we are showing the error if the old password is not correct.

File: itez/users/api/serializers.py (hunk 2 above)
Comment (jmnda-dev) [reply itez/users/api/serializers.py:44]: Yes, we compare `password` and `password2(confirm password)` if they are equal.

**Pull request merged**